src/local-ssh/proxy.ts

```diff
@@ -133,14 +129,37 @@ class WebSocketSSHProxy {
         session.credentials.publicKeys.push(keys);
 
         let pipePromise: Promise<void> | undefined;
-        session.onAuthenticating((e) => {
-            e.authenticationPromise = this.authenticateClient(e.username ?? '')
-                .then(pipeSession => {
+        session.onAuthenticating(async (e) => {
+
+            const flow: SSHUserFlowTelemetry = {
+                gitpodHost: this.options.host,
+                workspaceId: e.username ?? '',
+                daemonVersion: getDaemonVersion(),
+            };
+            this.extensionIpc.sendLocalSSHUserFlowStatus({ flowStatus: 'connecting', ...flow }).catch(e => {
+                this.logService.error(e, 'failed to send connecting flow status');
+            });
+            e.authenticationPromise = this.authenticateClient(e.username ?? '', flow)
+                .then(async pipeSession => {
+                    this.extensionIpc.sendLocalSSHUserFlowStatus({ flowStatus: 'connected', ...flow }).then().catch(e => {
+                        this.logService.error(e, 'failed to send connected flow status');
+                    });
                     pipePromise = session.pipe(pipeSession);
                     return {};
-                }).catch(async error => {
-                    this.logService.error(error, 'failed to authenticate client with username: ' + e.username);
-                    await session.close(SshDisconnectReason.byApplication, error.toString(), error instanceof Error ? error : undefined);
+                }).catch(async err => {
+                    await Promise.allSettled([
+                        (async () => {
+                            if (err instanceof FailedToProxyError) {
+                                flow.flowFailureCode = err.failureCode;
+                            }
+                            await this.extensionIpc.sendLocalSSHUserFlowStatus({ flowStatus: 'failed', ...flow, failureCode: err.failureCode }).catch(e => {
+                                this.logService.error(e, 'failed to send failed flow status');
+                            });
+                        })(),
+                        this.sendErrorReport(flow, err, 'failed to authenticate proxy with username: ' + e.username)
+                    ]);
+                    this.logService.error(err, 'failed to authenticate client with username: ' + e.username);
+                    await session.close(SshDisconnectReason.byApplication, err.toString(), err instanceof Error ? err : undefined);
                     return null;
                 });
         });
```

Comment:
@mustard-mh what is about code below? what is try - catch there?

Reply:
> what is try - catch there?

What do you mean? the try catch with `this.authenticateClient`? it's piping the results, return `{}` means accept, `null` means reject if you check document of `e.authenticationPromise`

Reply:
> what is about code below?

I have no idea what it (catch) will be after this [hack code](https://github.com/gitpod-io/gitpod-vscode-desktop/blob/2271e81660378696bbd66605390b7a84943f0f03/src/local-ssh/proxy.ts#L114-L122) need to test it

Reply:
I mean do we need to cover https://github.com/gitpod-io/gitpod-vscode-desktop/blob/2271e81660378696bbd66605390b7a84943f0f03/src/local-ssh/proxy.ts#L166-L176 or it is not related to initial connection?

we for sure should collect or exceptions in GCP since we don't have logging

Reply:
I have no idea how to trigger it, `gp stop` will not make produce `this.logService.error(e, 'failed to connect to client'); ` logs.

> or it is not related to initial connection?
It's after connection, to pipe stream

> we for sure should collect or exceptions in GCP since we don't have logging
Yes we can report error here

Reply:
> We will remove logging and put proper error reporting instead in another PR or this one?

We're not logging anything, it's a `NopeLogger`, when we want to debug (get logs), we will use `DebugLogger` but it will not in production

Reply:
Addressed, feel free to resolve conversation or have further common 🙏

Reply:
ok, then let's go and check that all error logs places or catches report to GCP in this PR 🙏  (in local proxy)

Reply:
> ok, then let's go and check that all error logs places or catches report to GCP in this PR 🙏 (in local proxy)

They are 👍